### PR TITLE
pricing - repair sessions started the 7th dec and terminated the day after

### PR DESCRIPTION
**Pull request by ClaudeROSSI**
Migration task to fix sessions that have been priced at zero because they were started with the "simple pricing logic" and terminated with the "built-in pricing engine".

This mainly impact the sessions of the slfcah (charge at home). But some other tenants may have the same issue.
The migration task simply price again the transaction and the corresponding consumption data.

### CODE REVIEW

File: src/migration/tasks/RepairTransactionPricedAtZeroTask.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+import TenantMigrationTask from '../TenantMigrationTask';
+import Transaction from '../../types/Transaction';
+import TransactionStorage from '../../storage/mongodb/TransactionStorage';
+import Utils from '../../utils/Utils';
+import chalk from 'chalk';
+
+const TASK_NAME = 'RepairTransactionPricedAtZeroTask';
+
+export default class RepairTransactionPricedAtZero extends TenantMigrationTask {
+
```
Comment (LucasBrazi06) [inline src/migration/tasks/RepairTransactionPricedAtZeroTask.ts:19]: BL

File: src/migration/tasks/RepairTransactionPricedAtZeroTask.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  pricingSettings: PricingSettings;
+
+  async loadSimplePricingSettings(tenant: Tenant): Promise<void> {
```
Comment (LucasBrazi06) [inline src/migration/tasks/RepairTransactionPricedAtZeroTask.ts:22]: visibility missing (public?)

File: src/migration/tasks/RepairTransactionPricedAtZeroTask.ts [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+            'stop.totalConsumptionWh': { $gt: 0 } ,
+            'stop.pricingSource': 'simple',
+            'refundData': { $exists: false }
+          }
+        },
+        {
+          $project: { '_id': 1 }
+        }
+      ]).toArray();
+
```
Comment (LucasBrazi06) [inline src/migration/tasks/RepairTransactionPricedAtZeroTask.ts]: BL

File: src/migration/tasks/RepairTransactionPricedAtZeroTask.ts [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+import Logging from '../../utils/Logging';
+import { ServerAction } from '../../types/Server';
+import SettingStorage from '../../storage/mongodb/SettingStorage';
+import TenantMigrationTask from '../TenantMigrationTask';
+import Transaction from '../../types/Transaction';
+import TransactionStorage from '../../storage/mongodb/TransactionStorage';
+import Utils from '../../utils/Utils';
+import chalk from 'chalk';
+
+const TASK_NAME = 'RepairTransactionPricedAtZeroTask';
```
Comment (LucasBrazi06) [inline src/migration/tasks/RepairTransactionPricedAtZeroTask.ts:16]: No constant in other classes, no constant in this one either. Or change all the other classes ;-)

File: src/migration/tasks/RepairTransactionPricedAtZeroTask.ts [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+        }
+      ]).toArray();
+    // Load Simple Pricing Settings
+    await this.loadSimplePricingSettings(tenant);
+    if (transactionsMDB.length > 0 && this.pricingSettings?.simple?.price > 0) {
+      let message = `${transactionsMDB.length} Transaction(s) are going to be repaired in Tenant ${Utils.buildTenantName(tenant)}...`;
+      await Logging.logInfo({
+        tenantID: Constants.DEFAULT_TENANT,
+        action: ServerAction.MIGRATION,
+        module: TASK_NAME, method: 'migrateTenant',
```
Comment (LucasBrazi06) [inline src/migration/tasks/RepairTransactionPricedAtZeroTask.ts:90]: Use MODULE_NAME constants like the other classes.

File: src/migration/tasks/RepairTransactionPricedAtZeroTask.ts (hunk 1 above)
Comment (ClaudeROSSI) [reply src/migration/tasks/RepairTransactionPricedAtZeroTask.ts]: ok

File: src/migration/tasks/RepairTransactionPricedAtZeroTask.ts (hunk 2 above)
Comment (ClaudeROSSI) [reply src/migration/tasks/RepairTransactionPricedAtZeroTask.ts]: ok - private

File: src/migration/tasks/RepairTransactionPricedAtZeroTask.ts (hunk 3 above)
Comment (ClaudeROSSI) [reply src/migration/tasks/RepairTransactionPricedAtZeroTask.ts]: ok

File: src/migration/tasks/RepairTransactionPricedAtZeroTask.ts (hunk 4 above)
Comment (ClaudeROSSI) [reply src/migration/tasks/RepairTransactionPricedAtZeroTask.ts]: rigth

File: src/migration/tasks/RepairTransactionPricedAtZeroTask.ts (hunk 5 above)
Comment (ClaudeROSSI) [reply src/migration/tasks/RepairTransactionPricedAtZeroTask.ts]: ok

File: src/migration/tasks/RepairTransactionPricedAtZeroTask.ts [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+import TransactionStorage from '../../storage/mongodb/TransactionStorage';
+import Utils from '../../utils/Utils';
+import chalk from 'chalk';
+
+const MODULE_NAME = 'RepairTransactionPricedAtZero';
+
+export default class RepairTransactionPricedAtZero extends TenantMigrationTask {
+  pricingSettings: PricingSettings;
+
+  private async loadSimplePricingSettings(tenant: Tenant): Promise<void> {
```
Comment (LucasBrazi06) [inline src/migration/tasks/RepairTransactionPricedAtZeroTask.ts:22]: private at the end, normally eslint should have yelled :-)

**Review by LucasBrazi06 — COMMENTED** [review]
One quick question, do you think we can use this also to recompute any transactions' pricing?

**Pull request merged**